src/java.base/share/classes/java/nio/file/CopyMoveHelper.java

```diff
@@ -109,21 +109,21 @@ static void copyToForeignTarget(Path source, Path target,
         LinkOption[] linkOptions = (opts.followLinks) ? new LinkOption[0] :
             new LinkOption[] { LinkOption.NOFOLLOW_LINKS };
 
-        // retrieve source posix view, null if unsupported
-        final PosixFileAttributeView sourcePosixView =
-            Files.getFileAttributeView(source, PosixFileAttributeView.class);
+        // determine whether the source supports posix attributes
+        boolean sourceSupportsPosixAttributes = Files.getFileAttributeView
+            (source, PosixFileAttributeView.class) != null;
 
         // attributes of source file
         BasicFileAttributes sourceAttrs = null;
-        if (sourcePosixView != null) {
+        if (sourceSupportsPosixAttributes)
             sourceAttrs = Files.readAttributes(source,
                                                PosixFileAttributes.class,
                                                linkOptions);
-        }
         if (sourceAttrs == null)
```

Comment:
This could if-then-else too.

Reply:
Right, but it can be simplified to `if (sourceSupportsPosixAttributes) { .. } else { ..}`.  What you have is okay too.